File: app/src/main/java/com/ty/voogla/net/gson/IntegerDefault0Adapter.java
package com.ty.voogla.net.gson;

import com.google.gson.*;

import java.lang.reflect.Type;

/**
 * desc:int值的类型转换器
 * author: XingZheng
 * date:2016/12/1
 * @author TY
 */

public class IntegerDefault0Adapter implements JsonSerializer<Integer>, JsonDeserializer<Integer> {
    @Override
    public Integer deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context)
            throws JsonParseException {
        try {
            //定义为int类型,如果后台返回""或者null,则返回0
            if (json.getAsString().equals(" ") ||json.getAsString().equals("") || json.getAsString().equals("null")) {
                return 0;
            }
        } catch (Exception ignore) {
        }
        try {
            return json.getAsInt();
        } catch (NumberFormatException e) {
            throw new JsonSyntaxException(e);
        }
    }

    @Override
    public JsonElement serialize(Integer src, Type typeOfSrc, JsonSerializationContext context) {
        return new JsonPrimitive(src);
    }
}

File: app/src/main/java/com/ty/voogla/net/gson/IntegerDefault0Adapter.kt
package com.ty.voogla.net.gson

import com.google.gson.*

import java.lang.reflect.Type

/**
 * desc:int值的类型转换器
 * author: XingZheng
 * date:2016/12/1
 * @author TY
 */

class IntegerDefault0Adapter : JsonSerializer<Int>, JsonDeserializer<Int> {
    @Throws(JsonParseException::class)
    override fun deserialize(json: JsonElement, typeOfT: Type, context: JsonDeserializationContext): Int? {
        try {
            //定义为int类型,如果后台返回""或者null,则返回0
            if (json.asString == " " || json.asString == "" || json.asString == "null") {
                return 0
            }
        } catch (ignore: Exception) {
        }

        try {
            return json.asInt
        } catch (e: NumberFormatException) {
            throw JsonSyntaxException(e)
        }

    }

    override fun serialize(src: Int?, typeOfSrc: Type, context: JsonSerializationContext): JsonElement {
        return JsonPrimitive(src!!)
    }
}